### ISSUE TITLE
Why don't the interstate or major road route numbers show on any zoom level?

### ISSUE THREAD

**Issue by fnicollet** [mapbox/mapbox-gl-js#962]
Hello,

I am wondering if this is the expected behaviour or some kind of bug in label placement. A good example is the city of Cleveland, Ohio:
http://www.openstreetmap.org/relation/182130#map=14/41.4887/-81.6851
In the OSM data, the highway references are present. But when rendered via mapbox-gl-js, there is no interstate label (I believe americans use "shields" to displayed the interstate names).
The strange thing is that the "Warden" style of Mapbox (raster tiles) show those interstate references so I guess the interstate data is already in the tiles somehow.
Here is what it looks like with mapbox-gl-js:
![image](https://cloud.githubusercontent.com/assets/1240481/5949260/2e3bb48a-a74c-11e4-9359-4eb2cdb77bfb.png)

Could this be a bug in the label placement? Or is it by design in bright-v6.json (like the missing housenumbers)?
I see there is a style for it in bright-v6.json:
 ```javascript
{
    "id": "road_label_highway_shields",
    "type": "symbol",
    "source": "mapbox",
    "source-layer": "road_label",
    "filter": ["all", ["==", "class", "motorway"], ["in", "reflen", 1, 2, 3, 4, 5, 6]],
    "layout": {
      "symbol-placement": "line",
      "symbol-min-distance": 500,
      "icon-image": "motorway_{reflen}",
      "icon-max-size": 1,
      "text-field": "{ref}",
      "text-font": "@sans_bd",
      "text-max-size": 11,
      "text-rotation-alignment": "viewport",
      "icon-rotation-alignment": "viewport"
    },
    "paint": {
      "text-color": "#765",
      "text-size": 11
    }
  }
```

Please note that even with bright-v6.json, some highway shields are placed. Like for instance near Lyon, France, you can see this:
![image](https://cloud.githubusercontent.com/assets/1240481/5949517/c7e7deb8-a74e-11e4-9216-0f640424d182.png)

Thanks for your help,
Fabien

**Comment by jfirebaugh** [comment]
This was fixed by #989.

**Comment by fnicollet** [comment]
Hello,

Did you try it on Cleveland after #989 ? Just built a version with the latest changes and It doesn't seem to change anything:
Before
![image](https://cloud.githubusercontent.com/assets/1240481/6177286/a6c8f9d4-b305-11e4-9ceb-36f0adc7b062.png)
After
![image](https://cloud.githubusercontent.com/assets/1240481/6177295/b4630bfc-b305-11e4-9759-b1d06dd8ae03.png)

Labels are the same, if you overlay the two images, you can see that it is slightly different, the POI icons are a little offset but that's it.
@ansis seemed to want to revert (https://github.com/mapbox/mapbox-gl-js/pull/989#issuecomment-73814279), is that one bad example?

Fabien

**Comment by fnicollet** [comment]
Nevermind, I actually didn't update the right branch or something, I am not on master but on mb-pages so I don't have this commit yet

**Comment by fnicollet** [comment]
Version 0.7 brings better results but some highway labels are offset by a pixel which creates a line:
![image](https://cloud.githubusercontent.com/assets/1240481/6537951/2ca25a34-c45c-11e4-98fc-38e98c59c88d.png)

I'll open another issue for this.
